Microsoft.WindowsAzure.Storage/src/cloud_blob.cpp

```diff
@@ -803,15 +828,13 @@ namespace azure { namespace storage {
                                         }
                                         condition_variable->notify_all();
                                         pplx::details::atomic_decrement(writer);
-                                        if (!released)
-                                        {
-                                            semaphore->unlock();
-                                        }
                                     }
                                 }
                             });
                         });
+                        parallel_tasks.emplace_back(std::move(parallel_task));
                     }
+                    // Code below is nonsense, becasuse exceptions won't be thrown from wait_all_async.
```

Comment:
Then I think it is proper to delete these code.

Reply:
Yeah, but that is legacy code from ancient times, dare we just delete it?

Reply:
I think a comment like this here is indicating that the code is useless, hence we should be able to safely delete it.

Reply:
I think it indicates that the way exceptions being handled is wrong, but doesn't in any sense suggest the semaphore operation is useless.

Reply:
If it bothers you that much, I'll just delete it.

Reply:
Done